packages/astro/src/core/middleware/sequence.ts

```diff
@@ -49,6 +51,22 @@ export function sequence(...handlers: MiddlewareHandler[]): MiddlewareHandler {
 							payload,
 							handleContext.request,
 						);
+
+						// This is a case where the user tries to rewrite from a SSR route to a prerendered route (SSG).
+						// This case isn't valid because when building for SSR, the prerendered route disappears from the server output because it becomes an HTML file,
+						// so Astro can't retrieve it from the emitted manifest.
+						if (
```

Comment:
Is there a way we could avoid the duplication across all the pipelines?

Reply:
It really depends on how you want to avoid the duplication. We can put the check in one single function, but I think the error should be thrown where we consume the check. What do you have in mind?